Error when filtering by Users.  "AttributeError: 'NoneType' object has no attribute 'users'"

**Description of problem**

Under the menu item: "Jobs" if we select Filters and then "Users" we get the following and no data is shown;

![image](https://github.com/user-attachments/assets/5442b431-4d8a-4eb6-863e-b4b682c37e1c)

`/var/log/messages` reports this

```
10:02:51 headnode_1 uwsgi[2414774]: uWSGIWorker4Core0: [ERROR] Exception on /api/users [GET]
Aug  6 10:02:51 headnode_1 uwsgi[2414774]: Traceback (most recent call last):
Aug  6 10:02:51 headnode_1 uwsgi[2414774]:  File "/usr/lib/python3.9/site-packages/flask/app.py", line 2073, in wsgi_app
Aug  6 10:02:51 headnode_1 uwsgi[2414774]:    response = self.full_dispatch_request()
Aug  6 10:02:51 headnode_1 uwsgi[2414774]:  File "/usr/lib/python3.9/site-packages/flask/app.py", line 1518, in full_dispatch_request
Aug  6 10:02:51 headnode_1 uwsgi[2414774]:    rv = self.handle_user_exception(e)
Aug  6 10:02:51 headnode_1 uwsgi[2414774]:  File "/usr/lib/python3.9/site-packages/flask/app.py", line 1516, in full_dispatch_request
Aug  6 10:02:51 headnode_1 uwsgi[2414774]:    rv = self.dispatch_request()
Aug  6 10:02:51 headnode_1 uwsgi[2414774]:  File "/usr/lib/python3.9/site-packages/flask/app.py", line 1502, in dispatch_request
Aug  6 10:02:51 headnode_1 uwsgi[2414774]:    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
Aug  6 10:02:51 headnode_1 uwsgi[2414774]:  File "/usr/lib/python3.9/site-packages/rfl/web/tokens.py", line 55, in wrapped
Aug  6 10:02:51 headnode_1 uwsgi[2414774]:    return view(*args, **kwargs)
Aug  6 10:02:51 headnode_1 uwsgi[2414774]:  File "/usr/lib/python3.9/site-packages/slurmweb/views/gateway.py", line 159, in users
Aug  6 10:02:51 headnode_1 uwsgi[2414774]:    for user in current_app.authentifier.users()
Aug  6 10:02:51 headnode_1 uwsgi[2414774]: AttributeError: 'NoneType' object has no attribute 'users'
Aug  6 10:02:51 headnode_1 uwsgi[2414774]: [pid: 2414774|app: 0|req: 193/968] 172.21.205.12 () {76 vars in 1524 bytes} [Tue Aug  6 10:02:51 2024] GET /api/users => generated 215 bytes in 0 msecs (HTTP/1.1 500) 2 headers in 91 bytes (2 switches on core 0)
```

Other Filters work as expected, it's just Users that shows an error. 

We thought it might be because we didn't have any users in Slurm's Db (we use AD for users to authenticate with), so we tried adding a normal user into slurmdbd on our TEST cluster, and that didn't help any.

We're using htttpd (apache2) as a reverse proxy for slurm-web with https passing to the local socks, and this seems to be working fine.

Using our TEST cluster to diagnose this. One headnode, 4 fairly beefy (40 cores, 384Gb each) work nodes with 13 partitions. 

These are new installs, we have not previously looks at Slurm-Web and have not had this Filter working before.

**Versions**

Running on Rocky Linux 9.3, software versions:
```
python3-slurm-web.noarch                    3.1.0-1.el9                      @rackslab
slurm.x86_64                                23.02.5-1.el9                    @@commandline
slurm-perlapi.x86_64                        23.02.5-1.el9                    @@commandline
slurm-slurmctld.x86_64                      23.02.5-1.el9                    @@commandline
slurm-slurmd.x86_64                         23.02.5-1.el9                    @@commandline
slurm-slurmdbd.x86_64                       23.02.5-1.el9                    @@commandline
slurm-slurmrestd.x86_64                     23.02.5-1.el9                    @@commandline
slurm-web-agent.noarch                      3.1.0-1.el9                      @rackslab
slurm-web-gateway.noarch                    3.1.0-1.el9                      @rackslab
```


Thank you for any thoughts.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -22,6 +22,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ### Fixed
 - backend: handle rfl.settings.errors.SettingsSiteLoaderError when loading
   gateway and backend site configuration (#317).
+- gateway: Response with HTTP/501 and JSON error when requesting users with
+  authentication disabled.
 - agent:
   - Translate HTTP/404 from slurmrestd into JSON error agent that can be
     interpreted by frontend and emit clear error message in logs (#321).

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -36,6 +36,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
     (vue-template-compiler).
   - Display empty list of users/account with light gray cross instead of dot in
     reservations page (#336).
+  - Hide users disclosure from jobs filters panel when authentication is
+    disabled (#330).
 
 ## [3.1.0] - 2024-07-03
 

--- a/frontend/src/components/jobs/JobsFiltersPanel.vue
+++ b/frontend/src/components/jobs/JobsFiltersPanel.vue
@@ -8,6 +8,7 @@
 
 <script setup lang="ts">
 import { useRuntimeStore } from '@/stores/runtime'
+import { useRuntimeConfiguration } from '@/plugins/runtimeConfiguration'
 import PartitionFilterSelector from '@/components/filters/PartitionFilterSelector.vue'
 import UserFilterSelector from '@/components/jobs/UserFilterSelector.vue'
 import AccountFilterSelector from '@/components/jobs/AccountFilterSelector.vue'
@@ -44,6 +45,7 @@ const props = defineProps({
 })
 
 const runtimeStore = useRuntimeStore()
+const runtimeConfiguration = useRuntimeConfiguration()
 
 const state_filters = [
   { value: 'completed', label: 'Completed' },
@@ -144,7 +146,17 @@ const state_filters = [
                   </div>
                 </DisclosurePanel>
               </Disclosure>
-              <Disclosure as="div" class="border-t border-t-gray-200 px-4 py-6" v-slot="{ open }">
+              <!--
+                Hide users filters disclosure panel when authentication is disabled. The list of
+                users are retrieved from authentication backend. When authentication is disabled,
+                the list of users cannot be retrieved.
+              -->
+              <Disclosure
+                v-if="runtimeConfiguration.authentication"
+                as="div"
+                class="border-t border-t-gray-200 px-4 py-6"
+                v-slot="{ open }"
+              >
                 <h3 class="-mx-2 -my-3 flow-root">
                   <DisclosureButton
                     class="flex w-full items-center justify-between bg-white px-2 py-3 text-sm text-gray-400"

--- a/slurmweb/apps/__init__.py
+++ b/slurmweb/apps/__init__.py
@@ -97,7 +97,7 @@ class SlurmwebWebApp(SlurmwebGenericApp, Flask):
         self.debug_flags = seed.debug_flags
 
         # register generic error handler
-        for error in [401, 403, 404, 500]:
+        for error in [401, 403, 404, 500, 501]:
             self.register_error_handler(error, self._handle_bad_request)
 
     def _handle_bad_request(self, error):

--- a/slurmweb/views/gateway.py
+++ b/slurmweb/views/gateway.py
@@ -153,6 +153,12 @@ def clusters():
 
 @check_jwt
 def users():
+    # If authentication is disabled, the list of users cannot be retrieved. Respond with
+    # HTTP/501 (not implemented) with a descriptive JSON error.
+    if current_app.authentifier is None:
+        err = "Unable to retrieve users when authentication is disabled"
+        logger.warning(err)
+        abort(501, err)
     return jsonify(
         [
             {"login": user.login, "fullname": user.fullname}

